api/src/test/java/org/apache/iceberg/util/RandomUtil.java

```diff
@@ -246,15 +246,15 @@ public static List<Object> generateList(
       if (list.isElementOptional() && random.nextInt(20) == 1) {
         result.add(null);
       } else {
-        result.add(elementResult.get());
+        result.add(elements.get());
       }
     }
 
     return result;
   }
 
   public static Map<Object, Object> generateMap(
-      Random random, Types.MapType map, Supplier<Object> keyResult, Supplier<Object> valueResult) {
+      Random random, Types.MapType map, Supplier<Object> keyResult, Supplier<Object> values) {
```

Comment:
Similarly, I think `valueSupplier` is a better name. This also doesn't address the problem for `keyResult`, which should also be `keySupplier`.